torch/distributed/_sharded_tensor/ops/_common.py

```diff
@@ -120,15 +123,20 @@ def _result_distribute_with_col_rearrange(
     # Compute output splits
     split_size = get_split_size(sharding_dim_size, world_size)
     output_split_sizes = [0] * world_size
+    output_tensor_list = [None] * world_size
     for idx, placement in enumerate(weight._sharding_spec.placements):
-        output_split_sizes[placement.rank()] = get_chunked_dim_size(
-            sharding_dim_size, split_size, idx
+        output_split_size = get_chunked_dim_size(sharding_dim_size, split_size, idx)
+        output_split_sizes[placement.rank()] = output_split_size
+        dims[0] = output_split_size
+        output_tensor_list[placement.rank()] = torch.empty(
+            *dims,
+            device=input.device,
+            dtype=combined_results.dtype,
         )
 
     # distribute the outputs using all2all.
-    dist.all_to_all_single(
-        output, combined_results, output_split_sizes=output_split_sizes, group=pg
-    )
+    output_tensor_list = all_to_all(results, group=pg, out_tensor_list=output_tensor_list)
```

Comment:
Isn't this going to cause a problem since we no longer specify output_split_sizes? Also, it doesn't look like `all_to_all` has an `out_tensor_list` argument?

Reply:
@pritamdamania87 I added one here because we don't support that. The output split size is specified by creating the `output_tensor_list`